crates/ntx-builder/src/coordinator.rs

```diff
@@ -0,0 +1,273 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use anyhow::Context;
+use indexmap::IndexMap;
+use miden_node_proto::domain::account::NetworkAccountPrefix;
+use miden_node_proto::domain::mempool::MempoolEvent;
+use miden_node_proto::domain::note::NetworkNote;
+use miden_objects::transaction::TransactionId;
+use tokio::sync::mpsc::error::SendError;
+use tokio::sync::{Semaphore, mpsc};
+use tokio::task::JoinSet;
+use tokio_util::sync::CancellationToken;
+
+use crate::actor::{AccountActor, AccountActorContext, AccountOrigin, ActorShutdownReason};
+
+// ACTOR HANDLE
+// ================================================================================================
+
+/// Handle to account actors that are spawned by the coordinator.
+#[derive(Clone)]
+struct ActorHandle {
+    event_tx: mpsc::Sender<Arc<MempoolEvent>>,
+    cancel_token: CancellationToken,
```

Comment:
I think this is redundant. The actor should already cancel if the channel gets closed so that's essentially the same thing.